iotdb-core/metrics/interface/src/main/java/org/apache/iotdb/metrics/AbstractMetricService.java

```diff
@@ -259,29 +259,45 @@ public Timer getOrCreateTimerWithInternalReport(
   /** Count with internal report. */
   public void countWithInternalReportAsync(
       long delta, String metric, MetricLevel metricLevel, long time, String... tags) {
-    internalReporter.writeMetricToIoTDB(
-        metricManager.count(delta, metric, metricLevel, tags), metric, time, tags);
+    if (metricManager.invalid(metricLevel, metric, tags)) {
```

Comment:
Do not use invalid here, may be we can use removeCounter

Reply:
This function writes directly to iotdb and uses the gauge type. I don't think it's necessary to computeIfAbsent a new object every time in metrics and never release it. The memory leak we have found so far is due to this. Even if using removeCounter solves the problem, why register with an unrelated place? prometheus doesn't need to know that

<img width="1039" alt="image" src="https://github.com/user-attachments/assets/ad5e99ce-9d49-4083-b5ae-b3ad151a0572" />